homeassistant/components/elgato/light.py

```diff
@@ -143,12 +143,15 @@ async def async_update(self) -> None:
             self._state = None
 
     @property
-    def device_info(self) -> dict[str, Any]:
+    def device_info(self) -> DeviceInfo | None:
         """Return device information about this Elgato Key Light."""
-        return {
-            ATTR_IDENTIFIERS: {(DOMAIN, self._info.serial_number)},
-            ATTR_NAME: self._info.product_name,
-            ATTR_MANUFACTURER: "Elgato",
-            ATTR_MODEL: self._info.product_name,
-            ATTR_SOFTWARE_VERSION: f"{self._info.firmware_version} ({self._info.firmware_build_number})",
-        }
+        return cast(
```

Comment:
The cast should not be needed here

Reply:
`homeassistant/components/elgato/light.py:149: error: Expected TypedDict key to be string literal  [misc]`

Reply:
Its a typed dict now so you have to change it to be the literal strings.